experiments/GCM/heldsuarez.jl

```diff
@@ -0,0 +1,140 @@
+                                     ref_state=HydrostaticState(IsothermalProfile(setup.T_initial), FT(0)),
+                                     turbulence=ConstantViscosityWithDivergence(FT(0)),
+                                     moisture=DryModel(),
+                                     sources=(Gravity(),
+                                              Coriolis(),
+                                              held_suarez_forcing!))
+    CLIMA.GCM_SetupGrid!(config, N, resolution, setup.domain_height)
+
+    return config
+end
+
+function held_suarez_forcing!(source, state, aux, t::Real)
+    FT = eltype(state)
+
+    ρ     = state.ρ
+    ρu    = state.ρu
+    ρe    = state.ρe
+    coord = aux.coord
+    Φ     = aux.orientation.Φ
+    e     = ρe / ρ
+    u     = ρu / ρ
+    e_int = e - u' * u / 2 - Φ
+    T     = air_temperature(e_int)
+
+    # Held-Suarez constants
+    k_a       = FT(1 / (40 * day))
+    k_f       = FT(1 / day)
+    k_s       = FT(1 / (4 * day))
+    ΔT_y      = FT(60)
+    Δθ_z      = FT(10)
+    T_equator = FT(315)
+    T_min     = FT(200)
+
+    σ_b          = FT(7 / 10)
+    r            = norm(coord, 2)
+    @inbounds λ  = atan(coord[2], coord[1])
+    @inbounds φ  = asin(coord[3] / r)
+    h            = r - FT(planet_radius)
+    scale_height = FT(7000) # from Smolarkiewicz JAS 2001 paper
+    σ            = exp(-h / scale_height)
+
+    # TODO: use
+    #  p = air_pressure(T, ρ)
+    #  σ = p/p0
+    exner_p       = σ ^ (R_d / cp_d)
+    Δσ            = (σ - σ_b) / (1 - σ_b)
+    height_factor = max(0, Δσ)
+    T_equil       = (T_equator - ΔT_y * sin(φ) ^ 2 - Δθ_z * log(σ) * cos(φ) ^ 2 ) * exner_p
+    T_equil       = max(T_min, T_equil)
+    k_T           = k_a + (k_s - k_a) * height_factor * cos(φ) ^ 4
+    k_v           = k_f * height_factor
+
+    source.ρu += -k_v * ρu
+    source.ρe += -k_T * ρ * cv_d * (T - T_equil) - k_v * ρu' * ρu / ρ
+end
+
+function main()
+    CLIMA.init()
+
+    FT = Float64
+
+    # DG polynomial order
+    N = 5
+
+    # Domain resolution
+    nelem_horz = 6
+    nelem_vert = 8
+    resolution = (nelem_horz, nelem_vert)
+
+    setup = HeldSuarezSetup{FT}()
+
+    # Set up the driver configuration
+    config = config_heldsuarez(FT, N, resolution, setup)
+
+    # Determine the time step
+    element_size = (setup.domain_height / nelem_vert)
+    acoustic_speed = soundspeed_air(FT(315))
+    lucas_magic_factor = 5
+    dt = lucas_magic_factor * min_node_distance(config.grid) / acoustic_speed
+
+    timeend = FT(60) # 400day
+    outputtime = 2day # TODO
+
+    # TODO: this needs Q!
+    #=
+    # Set up the filter callback
+    filterorder = 14
+    filter = ExponentialFilter(config.grid, 0, filterorder)
+    cbfilter = EveryXSimulationSteps(1) do
+        Filters.apply!(config.Q, 1:size(Q, 2), config.grid, filter)
+        nothing
+    end
+    =#
+
+    result = CLIMA.invoke(dt, FT(0), timeend, config, LSRK144NiegemannDiehlBusch;
```

Comment:
Can we move the time stepper specification to the configuration files, with the possibility of overriding the default there by something that can be specified in the driver? Also, we should make sure that the default for LES and GCM is a 1D IMEX method.